charms/kserve-controller/src/charm.py

```diff
@@ -54,13 +75,42 @@
 ]
 
 
+def parse_images_config(config: str) -> Dict:
+    """
+    Parse a YAML config-defined images list.
+
+    This function takes a YAML-formatted string 'config' containing a list of images
+    and returns a dictionaryrepresenting the images.
+
+    Args:
+        config (str): YAML-formatted string representing a list of images.
+
+    Returns:
+        Dict: A list of images.
+    """
+    error_message = (
+        f"Cannot parse a config-defined images list from config '{config}' - this"
+        "config input will be ignored."
+    )
+    if not config:
+        return []
+    try:
+        images = yaml.safe_load(config)
+    except yaml.YAMLError as err:
+        log.warning(f"{error_message}  Got error: {err}, while parsing the custom_image config.")
+        return []
```

Comment:
(I know where this code came from, so I accept the "but you did it this way too" argument :) )

For images, do we want this "if invalid, ignore" behaviour for custom images?  Or should we block the charm?  My first thought it that we block the charm, mainly because this is a bit of a security thing where we want custom images to assert we know they're safe.  A silent error here would result in a charm that seems to work as expected, but that is using all default images.  

wdyt?